manifests/repo.pp

```diff
@@ -12,15 +12,23 @@
 #
 # @param gpgkey
 #   The GPG key to use
+#
+# @param baseurl
+#   An optional base URL to be used for yumrepo, instead of the default
 class candlepin::repo (
-  String $version,
+  Variant[Undef, Enum['nightly'], Pattern['^\d+\.\d+$']] $version = undef,
   String $dist = "el${facts['os']['release']['major']}",
   Boolean $gpgcheck = false,
   Optional[String] $gpgkey = undef,
+  Optional[Stdlib::HTTPUrl] $baseurl = undef,
 ) {
+  unless $baseurl {
+    assert_type(NotUndef, $version)
+  }
+
   yumrepo { 'candlepin':
     descr    => 'Candlepin: an open source entitlement management system.',
-    baseurl  => "https://yum.theforeman.org/katello/${version}/candlepin/${dist}/\$basearch/",
+    baseurl  => pick($baseurl, "https://yum.theforeman.org/candlepin/${version}/${dist}/\$basearch/"),
```

Comment:
This also changes the base URL. Perhaps that should be its own commit or at least reflected in the commit message?